src/slider.jsx

```diff
@@ -428,9 +476,9 @@ const Slider = React.createClass({
 
   _updateWithChangeEvent(e, percent) {
     if (this.state.percent === percent) return;
-    this.setPercent(percent);
-    let value = this._alignValue(this._percentToValue(percent));
-    if (this.props.onChange) this.props.onChange(e, value);
+    this.setPercent(percent, () => {
+      if (this.props.onChange) this.props.onChange(e, this.state.value);
```

Comment:
Thanks @KapJI for review and for this finding. You are right and I just fixed it, but from setPercent, it seemed more convenient there for me.